### PR TITLE
Revised Test vectors text

### PR DESCRIPTION
**Pull request by Wind4Greg**
Removed "to be written" section. Cleaned up wording a bit.

***
<a href="https://pr-preview.s3.amazonaws.com/Wind4Greg/di-eddsa-2020/pull/24.html" title="Last updated on Feb 24, 2023, 6:19 PM UTC (6dc0118)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/vc-di-eddsa/24/5a3b66a...Wind4Greg:6dc0118.html" title="Last updated on Feb 24, 2023, 6:19 PM UTC (6dc0118)">Diff</a>

### CODE REVIEW

**Comment by iherman** [comment]
The issue was discussed in a [meeting](https://www.w3.org/2017/vc/WG/Meetings/Minutes/2023-02-08-vcwg#section3-6) on 2023-02-08

- no resolutions were taken

<details><summary><i>View the transcript</i></summary>

#### 3.6. Revised Test vectors text (pr vc-di-eddsa#24)

_See github pull request [vc-di-eddsa#24](https://github.com/w3c/vc-di-eddsa/pull/24)._




**Manu Sporny:** moving on to edwards curve suite.  
… p.r. to update test vectors.  
… good job, but not a working group member.  
… doesn't look like any ipr issues. needs to be cleared for IP.  
… thats it.  

</details>

**Review by msporny — APPROVED** [review]

File: index.html [hunk 1]
```diff
@@ -814,150 +814,156 @@ <h2>Privacy Considerations</h2>
...
-      </p>
-
-      <pre class="example">
+      <section>
+        <h3>Representation: Ed25519Signature2020</h3>
+        <p>
+The signer needs to generate a private/public key pair with the private key used 
+for signing and the public key made available for verification. Below the 
+[[MULTIBASE]]/[[MULTICODEC]] representation for the public key, <code>ed25519-pub</code>,
+and the representation for the private key, <code>ed25519-priv</code>, are shown.
```
Comment (TallTed) [inline index.html]: Shifted to (American) English word order.
```suggestion
for signing and the public key made available for verification. The 
[[MULTIBASE]]/[[MULTICODEC]] representation for the public key, <code>ed25519-pub</code>,
and the representation for the private key, <code>ed25519-priv</code>, are shown below.
```

File: index.html [hunk 2]
```diff
@@ -814,150 +814,156 @@ <h2>Privacy Considerations</h2>
...
+    publicKeyMultibase: "z6MkrJVnaZkeFzdQyMZu1cgjg7k1pZZ6pvBQ7XJPt4swbTQ2", 
+    privateKeyMultibase: "z3u2en7t5LR2WtQH5PfFqMqwVHBeXouLzo6haApm8XHqvjxq" 
 }
-      </pre>
-      <pre class="example">
+        </pre>
+
+        <p>
+Signing begins with a credential without an attached proof which is then converted 
+to canonical form and then hashed as shown in the following three examples.
```
Comment (TallTed) [inline index.html]:
```suggestion
Signing begins with a credential without an attached proof, which is converted 
to canonical form, and then hashed, as shown in the following three examples.
```

File: index.html [hunk 3]
```diff
@@ -814,150 +814,156 @@ <h2>Privacy Considerations</h2>
...
+      </pre>
+
+      <pre class="example" title="Hash of Canonical Credential without Proof (hex)">
+6c6b2795e7fa33a9fb28062527142b3c6edf7ba239942439b6f0bb0851b3cce3
       </pre>
 
-      <pre class="example">
+      <p>
+The next step is to take the proof options document, convert it to canonical form,
+and obtain its hash as shown in the next three examples.
```
Comment (TallTed) [inline index.html]:
```suggestion
and obtain its hash, as shown in the next three examples.
```

File: index.html [hunk 4]
```diff
@@ -814,150 +814,156 @@ <h2>Privacy Considerations</h2>
...
+_:c14n0 &lt;https://w3id.org/security#proofPurpose&gt; &lt;https://w3id.org/security#assertionMethod&gt; .
+_:c14n0 &lt;https://w3id.org/security#verificationMethod&gt; &lt;https://example.edu/issuers/565049#key-1&gt; .
+      </pre>
+
+      <pre class="example" title="Hash of Canonical Proof Options Document (hex)">
+565a2884ebb2d38aa34871108074ab51631ec812d33eb2473178bce19937ad09
+      </pre>
+
+      <p>
+Finally we combine the two hashes, use the private key with the combined hash to
```
Comment (TallTed) [inline index.html]:
```suggestion
Finally, we combine the two hashes, use the private key with the combined hash to
```

File: index.html [hunk 5]
```diff
@@ -814,150 +814,156 @@ <h2>Privacy Considerations</h2>
...
+
+      <p>Assemble the signed credential by:</p>
+      <ol>
+        <li>
+Adding the <code>proofValue</code> field with the previously computed base58-btc 
+value to the proof options document.
+        </li>
+        <li>
+Setting the <code>proof</code> field of the credential to the augmented proof 
+option document.
```
Comment (TallTed) [inline index.html]:
```suggestion
      <p>Assemble the signed credential with the following two steps:</p>
      <ol>
        <li>
Add the <code>proofValue</code> field with the previously computed base58-btc 
value to the proof options document.
        </li>
        <li>
Set the <code>proof</code> field of the credential to the augmented proof 
option document.
```

**Review by TallTed — CHANGES_REQUESTED** [review]
Small stuff.

File: index.html [hunk 6]
```diff
@@ -814,150 +814,156 @@ <h2>Privacy Considerations</h2>
...
+        <h3>Representation: Ed25519Signature2020</h3>
+        <p>
+The signer needs to generate a private/public key pair with the private key used 
+for signing and the public key made available for verification. Below the 
+[[MULTIBASE]]/[[MULTICODEC]] representation for the public key, <code>ed25519-pub</code>,
+and the representation for the private key, <code>ed25519-priv</code>, are shown.
+        </p>
+        <pre class="example" title="Private and Public keys for Signature">
 {
-  "seed_0": "9b937b81322d816cfab9d5a3baacc9b2a5febe4b149f126b3630f93a29527017"
```
Comment (dlongley) [inline index.html:823]: We should probably have `@context` here and `"type": "Multikey"`.

**Review by dlongley — APPROVED** [review]
LGTM as a clear improvement over what was previously there. I haven't checked the test vector against an implementation myself.

File: index.html [hunk 7]
```diff
@@ -814,150 +814,156 @@ <h2>Privacy Considerations</h2>
...
-    "proof": {
-      "type": "Ed25519Signature2020",
-      "created": "2019-12-11T03:50:55Z",
-      "proofValue": "z5SpZtDGGz5a89PJbQT2sgbRUiyyAGhhgjcf86aJHfYcfvPjxn6vej5na6kUzmw1jMAR9PJU9mowshQFFdGmDN14D",
-      "proofPurpose": "assertionMethod",
-      "verificationMethod": "https://example.com/issuer/123#key-0"
+  "@context": [
+    "https://www.w3.org/2018/credentials/v1",
+    "https://www.w3.org/2018/credentials/examples/v1",
+    "https://w3id.org/security/suites/ed25519-2020/v1"
```
Comment (msporny) [inline index.html:940]: Not for this PR, but a future PR should update this to use the v2 context (the example will look much cleaner, IMHO).

Comment (Wind4Greg) [reply index.html:940]: Will do. I'm currently working on test vectors for the "DataIntegrityProof" representation so will make sure to use the latest context!

File: index.html [hunk 8]
```diff
@@ -814,150 +814,156 @@ <h2>Privacy Considerations</h2>
...
-    "proof": {
-      "type": "Ed25519Signature2020",
-      "created": "2021-06-04T20:50:09Z",
-      "verificationMethod": "https://example.com/issuer/123#key-0",
-      "proofPurpose": "authentication",
-      "challenge": "123",
-      "proofValue": "z2y3UBXAiToXLzQqeMnHiMozJ3hKxcMgLm7p8GRQA92F6JSYu49RxHQf6k1CMKnMdpj3BLRSH69b9qA9cfjE3oS7q"
-    }
+  "proof": {
+    "type": "Ed25519Signature2020",
```
Comment (msporny) [inline index.html:957]: Also for a future PR, but we should switch to the use of `cryptosuite` and `eddsa-2022`.

Comment (Wind4Greg) [reply index.html:957]: Hmm, the type "Ed25519Signature2020" doesn't have a `cryptosuite` parameter. The type "DataIntegrityProof" does. I'm working on the type "DataIntegrityProof" still.

Comment (Wind4Greg) [reply index.html:957]: Hmm, the type "Ed25519Signature2020" doesn't have a `cryptosuite` parameter. The type "DataIntegrityProof" does. I'm working on the type "DataIntegrityProof" still.

Comment (msporny) [reply index.html:957]: Yep, sorry, I meant... switch to `type` of `DataIntegrityProof` with `cryptosuite` set to `eddsa-2022`.

File: index.html [hunk 9]
```diff
@@ -814,150 +814,156 @@ <h2>Privacy Considerations</h2>
...
+      "type": "BachelorDegree",
+      "name": "Bachelor of Science and Arts"
+    }
   }
 }
+        </pre>
+      
+        <pre class="example" title="Canonical Credential without Proof">
+          &lt;did:example:ebfeb1f712ebc6f1c276e12ec21&gt; &lt;https://example.org/examples#degree&gt; _:c14n0 .
+          &lt;http://example.edu/credentials/3732&gt; &lt;http://www.w3.org/1999/02/22-rdf-syntax-ns#type&gt; &lt;https://example.org/examples#UniversityDegreeCredential&gt; .
```
Comment (OR13) [inline index.html:863]: this escaping makes my eyes bleed, prefer a `data-include`

File: index.html [hunk 10]
```diff
@@ -814,150 +814,156 @@ <h2>Privacy Considerations</h2>
...
-    },
-    "proof": {
-      "type": "Ed25519Signature2020",
-      "created": "2019-12-11T03:50:55Z",
-      "proofValue": "z5SpZtDGGz5a89PJbQT2sgbRUiyyAGhhgjcf86aJHfYcfvPjxn6vej5na6kUzmw1jMAR9PJU9mowshQFFdGmDN14D",
-      "proofPurpose": "assertionMethod",
-      "verificationMethod": "https://example.com/issuer/123#key-0"
+  "@context": [
+    "https://www.w3.org/2018/credentials/v1",
+    "https://www.w3.org/2018/credentials/examples/v1",
```
Comment (OR13) [inline index.html:939]: I suggest your test vectors NOT require loading 3 other contexts....

File: index.html (hunk 9 above)
Comment (Wind4Greg) [reply index.html:863]: Thanks! I'm new to ReSpec. `data-include` should make bringing in the test vectors much easier.

File: index.html (hunk 10 above)
Comment (Wind4Greg) [reply index.html:939]: Agree! I posted an issue at https://github.com/w3c/vc-di-eddsa/issues/27. Looking for feedback on the example unsigned document (we've got three different test vector cases to produce). I was suggesting the following which only depends on one context. Also I posted the code that I'll use to generate the test vectors at https://github.com/Wind4Greg/EdDSA-Test-Vectors.

```json
{
    "@context": [
        "https://www.w3.org/ns/credentials/v2",
        {"UniversityDegreeCredential": "https://example.org/examples#UniversityDegreeCredential"}
    ],
    "id": "http://example.edu/credentials/3732",
    "type": [
        "VerifiableCredential",
        "UniversityDegreeCredential"
    ],
    "issuer": "https://example.edu/issuers/565049",
    "issuanceDate": "2010-01-01T00:00:00Z",
    "credentialSubject": {
        "id": "did:example:ebfeb1f712ebc6f1c276e12ec21",
        "degree": {
            "type": "BachelorDegree",
            "name": "Bachelor of Science and Arts"
        }
    }
}
```

File: index.html (hunk 9 above)
Comment (msporny) [reply index.html:863]: I think `data-include` breaks PR-preview, unless they recently fixed that... not saying we shouldn't use it, but something to consider.

File: index.html (hunk 6 above)
Comment (msporny) [reply index.html:823]: We should probably just break these values out and not have them in a JSON data structure that we don't want people producing in the real world. I'll do that after the merge. We haven't specified how / where to define `privateKeyMultibase` and we don't want that perma-thread holding up this PR.

File: index.html (hunk 10 above)
Comment (msporny) [reply index.html:939]: The VC above is problematic... `UniversityDegreeCredential` is defined, but the following items are going to be converted to issuer-defined properties, which might confuse some people: issuanceDate, degree, BachelorDegree. Let's keep the current examples and then simplify later. We should probably just shift to a simple hello world example using only the `credentials/v2` context... which does define properties like "name" and "description".

**Comment by msporny** [comment]
Editorial (test vectors are editorial), multiple reviews, changes requested (some made now, some will be made after the merge), no objections, merging.

File: index.html (hunk 10 above)
Comment (Wind4Greg) [reply index.html:939]: Sounds good.

**Pull request merged**